nomad/structs/network.go

```diff
@@ -36,13 +36,18 @@ var (
 
 // NetworkIndex is used to index the available network resources
 // and the used network resources on a machine given allocations
+//
+// Fields are exported so they may be JSON serialized for debugging.
+// Fields are *not* intended to be used directly.
 type NetworkIndex struct {
-	AvailNetworks  []*NetworkResource              // List of available networks
-	NodeNetworks   []*NodeNetworkResource          // List of available node networks
-	AvailAddresses map[string][]NodeNetworkAddress // Map of host network aliases to list of addresses
-	AvailBandwidth map[string]int                  // Bandwidth by device
-	UsedPorts      map[string]Bitmap               // Ports by IP
-	UsedBandwidth  map[string]int                  // Bandwidth by device
+	TaskNetworks  []*NetworkResource              // List of available networks
+	GroupNetworks []*NodeNetworkResource          // List of available node networks
```

Comment:
Are these values deprecated as well? `TaskNetworks` seems to be related to the network at `task -> resource` level and `GroupNetworks` I'm not even sure where they are used 😅 

If they are not, maybe update the comments as well?

```suggestion
	TaskNetworks  []*NetworkResource              // List of networks defined at the task level
	GroupNetworks []*NodeNetworkResource          // List of networks defined at the group level (???)
```

Reply:
Good call. Wrote some more books-in-comments. :grimacing: